tests/test_persistence.py

```diff
@@ -1633,6 +1345,50 @@ def test_save_on_flush_single_files(self, pickle_persistence, good_pickle_files)
             conversations_test = dict(pickle.load(f))['conversations']
         assert conversations_test['name1'] == conversation1
 
+    def test_custom_pickler_unpickler(
```

Comment:
why does this test use handler logic? Wouldn't it suffice to test with somethling like

```python
persistence.update_chat_data(chat_id, data_containing_bot)
persistence.flush()
persistence = Persistence(…)
loaded_data = persistence.get_chat_data
# make assertions
```
?

Reply:
> why use handler logic?

Wanted to be more sure that this would work, guess I can add this 'simpler' test in addition to the handle one

Reply:
I see. My reasoning is to separate the unit tests from the integration tests. `TestBasePersistence` already has tests for the dispatcher integration, so IMO it would suffice to test the methods of `PicklePersistence` here. I'm okay with having the handler-tests here as well, but in that case please make it a new test. That way, when dispatcher integration is change, it will be more clear why a failing test is failing :)